references/video_classification/train.py

```diff
@@ -0,0 +1,340 @@
+    dataset_test.video_clips.compute_clips(16, 1, frame_rate=15)
+    dataset_test.transform = torchvision.transforms.Compose([
+        T.ToFloatTensorInZeroOne(),
+        T.Resize((128, 171)),
+        T.Normalize((0.43216, 0.394666, 0.37645), (0.22803, 0.22145, 0.216989)),
+        T.CenterCrop((112, 112))
+    ])
+    """
+    if args.cache_dataset and os.path.exists(cache_path):
+        # Attention, as the transforms are also cached!
+        print("Loading dataset_test from {}".format(cache_path))
+        dataset_test, _ = torch.load(cache_path)
+    else:
+        dataset_test = torchvision.datasets.Kinetics(
+            valdir,
+            args.frames_per_clip,
+            1,
+            transforms.Compose([
+                transforms.Resize(256),
+                transforms.CenterCrop(224),
+                transforms.ToTensor(),
+                normalize,
+            ]))
+        if args.cache_dataset:
+            print("Saving dataset_test to {}".format(cache_path))
+            utils.mkdir(os.path.dirname(cache_path))
+            utils.save_on_master((dataset_test, valdir), cache_path)
+    """
+    print("Creating data loaders")
+    train_sampler = torchvision.datasets.video_utils.RandomClipSampler(dataset.video_clips, args.clips_per_video)
+    test_sampler = SequentialClipSampler(dataset_test.video_clips, args.clips_per_video)
+    if args.distributed:
+        train_sampler = DistributedSampler(train_sampler)
+        test_sampler = DistributedSampler(test_sampler)
+
+    data_loader = torch.utils.data.DataLoader(
+        dataset, batch_size=args.batch_size,
+        sampler=train_sampler, num_workers=args.workers, pin_memory=True)
+
+    data_loader_test = torch.utils.data.DataLoader(
+        dataset_test, batch_size=args.batch_size,
+        sampler=test_sampler, num_workers=args.workers, pin_memory=True)
+
+    print("Creating model")
+    # model = torchvision.models.video.__dict__[args.model](pretrained=args.pretrained)
+    model = torchvision.models.video.__dict__[args.model]()
+    model.to(device)
+    if args.distributed and args.sync_bn:
+        model = torch.nn.SyncBatchNorm.convert_sync_batchnorm(model)
+
+    criterion = nn.CrossEntropyLoss()
+
+    lr = args.lr * args.world_size
+    optimizer = torch.optim.SGD(
+        model.parameters(), lr=lr, momentum=args.momentum, weight_decay=args.weight_decay)
+
+    if args.apex:
+        model, optimizer = amp.initialize(model, optimizer,
+                                          opt_level=args.apex_opt_level
+                                          )
+
+    # per iteration, not per epoch
+    warmup_epochs = 10
+    warmup_iters = warmup_epochs * len(data_loader)
+    lr_milestones = [len(data_loader) * m for m in args.lr_milestones]
+    lr_scheduler = WarmupMultiStepLR(
+        optimizer, milestones=lr_milestones, gamma=args.lr_gamma,
+        warmup_iters=warmup_iters, warmup_factor=1e-5)
+
+    model_without_ddp = model
```

Comment:
Why two models?

Reply:
This is just a pointer to the other model without DDP, and is useful for loading / saving the checkpoint without having to worry if the model / checkpoint has DDP or not.

Reply:
That's a neat trick - is there a benefit to doing this as opposed to the utility that you use in `maskrcnn_benchmark`?

Reply:
This is just much simpler, and makes less assumptions, so I'm preferring this now over the implementation I had in `maskrcnn-benchmark`